### PR TITLE
Revised LP Encoding for Multi-objective verification under simple strategies 

### PR DESCRIPTION
**Pull request by tquatmann**
This is a revision of the corresponding LP encoding as presented in [Chapter 8 of my PhD Thesis](http://doi.org/10.18154/RWTH-2023-09669).

Also adds support for (qualitative and quantitative (aka numerical)) achievability queries.

I'm piggybacking support for gurobi version 11 here.

I still want to add a few tests for the new achievability functionality before this is ready to be merged.

### CODE REVIEW

File: src/storm/modelchecker/multiobjective/deterministicScheds/DeterministicSchedsParetoExplorer.cpp [hunk 1]
```diff
@@ -302,61 +302,42 @@ DeterministicSchedsParetoExplorer<SparseModelType, GeometryValueType>::Determini
...
-            }
-            STORM_PRINT_AND_LOG("#STATS " << nonConstMecCounter << " non-constant MECS in " << modelname << '\n');
-            STORM_PRINT_AND_LOG("#STATS " << nonConstMecStateCounter << " non-constant MEC States in " << modelname << '\n');
-            // Print the same statistics for the unfolded model as well.
-            modelname = "unfolded-model";
-        }
-        sw.stop();
-        STORM_PRINT_AND_LOG("#STATS " << sw << " seconds for computing these statistics.\n");
-    }
+    //    if (storm::settings::getModule<storm::settings::modules::CoreSettings>().isShowStatisticsSet()) {
```
Comment (sjunges) [inline src/storm/modelchecker/multiobjective/deterministicScheds/DeterministicSchedsParetoExplorer.cpp:313]: is this on purpose?

File: src/storm/modelchecker/multiobjective/deterministicScheds/DeterministicSchedsParetoExplorer.cpp [hunk 2]
```diff
@@ -407,12 +389,12 @@ std::unique_ptr<CheckResult> DeterministicSchedsParetoExplorer<SparseModelType,
                 storm::utility::vector::convertNumericVector<ModelValueType>(transformObjectiveValuesToOriginal(objectives, p.second.get())));
         }
     }
-    if (storm::settings::getModule<storm::settings::modules::CoreSettings>().isShowStatisticsSet()) {
-        STORM_PRINT_AND_LOG("#STATS " << paretoPoints.size() << " Pareto points\n");
-        STORM_PRINT_AND_LOG("#STATS " << unachievableAreas.size() << " unachievable areas\n");
-        STORM_PRINT_AND_LOG("#STATS " << overApproximation->getHalfspaces().size() << " unachievable halfspaces\n");
-        STORM_PRINT_AND_LOG(lpChecker->getStatistics("#STATS "));
-    }
+    //    if (storm::settings::getModule<storm::settings::modules::CoreSettings>().isShowStatisticsSet()) {
```
Comment (sjunges) [inline src/storm/modelchecker/multiobjective/deterministicScheds/DeterministicSchedsParetoExplorer.cpp:392]: and here

File: src/storm/modelchecker/multiobjective/deterministicScheds/VisitingTimesHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    }
+    return res;
+}
+
+template<typename ValueType>
+ValueType VisitingTimesHelper<ValueType>::computeMecVisitsUpperBound(storm::storage::MaximalEndComponent const& mec,
+                                                                     storm::storage::SparseMatrix<ValueType> const& transitions,
+                                                                     bool assumeOptimalTransitionProbabilities) {
+    auto const one = storm::utility::one<ValueType>();
+    auto const l = computeMecTraversalLowerBound(mec, transitions, assumeOptimalTransitionProbabilities);
```
Comment (sjunges) [inline src/storm/modelchecker/multiobjective/deterministicScheds/VisitingTimesHelper.cpp:63]: this variable l looks like a one below :-)

**Comment by sjunges** [comment]
LGTM, three minor comments to be considered.

**Comment by tquatmann** [comment]
Thanks!
I wasn't sure with the statistics output, but I think it's quite specific and generally not very useful.

**Comment by sjunges** [comment]
In the past, I have used ifdefs (only within the cpp file) for that, with the common understanding that this was code that I wanted to retain for the moment for debug purposes, but that was by no means stable or whatever. I don't know whether that is better.

**Pull request merged**